lib/players/computer.rb

```diff
@@ -0,0 +1,45 @@
+class Computer < Player
+  def move(board)
+    board.taken?(5) ? winning_move(board) : opening_move
+  end
+
+  def opening_move
+    "5"
+  end
+
+  def random_move(board)
+    (1..9).to_a.sample
+  end
+
+  def opponent
+    self.token == "X" ? "O" : "X"
+  end
+
+  def best_move(board)
+    (1..9).to_a.find { |move| !board.taken?(move) }
+  end
+
+  def block_move(board)
+    if !never_win(board, opponent).nil?
+      never_win(board, opponent).find {|location| board.cells[location] == " " } .to_i+1
+    end
+  end
+
+  def win(board)
+    if !never_win(board, self.token).nil?
+      never_win(board, self.token).find { |location| board.cells[location] ==  " " }.to_i+1
+    end
+  end
+
+  def winning_move(board)
+    block_move(board) || win(board) || best_move(board)
+  end
```

Comment:
Love the way that reads as a precedence of choices. However, shouldn't it be:

Try to win
Try to block
Pick a best move

?